integration-tests/vault/src/test/java/io/quarkus/vault/VaultITCase.java

```diff
@@ -0,0 +1,136 @@
+package io.quarkus.vault;
+
+import static io.quarkus.vault.CredentialProvider.PASSWORD_PROPERTY_NAME;
+import static io.quarkus.vault.CredentialProvider.USER_PROPERTY_NAME;
+import static io.quarkus.vault.VaultTestExtension.APP_SECRET_PATH;
+import static io.quarkus.vault.VaultTestExtension.DB_PASSWORD;
+import static io.quarkus.vault.VaultTestExtension.SECRET_KEY;
+import static io.quarkus.vault.VaultTestExtension.SECRET_PATH_V1;
+import static io.quarkus.vault.VaultTestExtension.SECRET_PATH_V2;
+import static io.quarkus.vault.VaultTestExtension.SECRET_VALUE;
+import static io.quarkus.vault.VaultTestExtension.VAULT_AUTH_USERPASS_PASSWORD;
+import static io.quarkus.vault.VaultTestExtension.VAULT_AUTH_USERPASS_USER;
+import static io.quarkus.vault.VaultTestExtension.VAULT_DBROLE;
+import static io.quarkus.vault.runtime.config.VaultAuthenticationType.USERPASS;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Map;
+import java.util.Properties;
+
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.jboss.logging.Logger;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.vault.runtime.VaultManager;
+import io.quarkus.vault.runtime.client.VaultClient;
+import io.quarkus.vault.runtime.client.dto.VaultDbCreds;
+import io.quarkus.vault.runtime.client.dto.VaultKvSecretV1;
+import io.quarkus.vault.runtime.client.dto.VaultKvSecretV2;
+import io.quarkus.vault.runtime.client.dto.VaultLeasesLookup;
+import io.quarkus.vault.runtime.client.dto.VaultLookupSelf;
+import io.quarkus.vault.runtime.client.dto.VaultRenewLease;
+import io.quarkus.vault.runtime.client.dto.VaultRenewSelf;
+import io.quarkus.vault.runtime.client.dto.VaultUserPassAuth;
+
+@DisabledOnOs(OS.WINDOWS)
+@ExtendWith(VaultTestExtension.class)
+public class VaultITCase {
```

Comment:
Is there a reason why we don't have a native test here?

Reply:
this goes back to your comment here https://github.com/quarkusio/quarkus/pull/2897/#discussion_r326097390

those tests are IT in nature because the launch a vault container and interact with it, though we are not building an application like in the vault-app integration-tests.
they were initially in modules agroal deployment and vault deployment modules.
integration-tests is for IT or native IT?